modules/codegen/src/main/scala/com/twilio/guardrail/generators/AkkaHttpServerGenerator.scala

```diff
@@ -443,44 +473,268 @@ object AkkaHttpServerGenerator {
           case x :: xs =>
             Some(xs.foldLeft[Term](x) { case (a, n) => q"${a} & ${n}" })
         }
-    }
 
-    def bodyToAkka(body: Option[ScalaParameter]): Target[Term] = {
+    def bodyToAkka(body: Option[ScalaParameter]): Target[Option[Term]] =
       Target.pure(
         body.map {
           case ScalaParameter(_, _, _, _, argType) =>
             q"entity(as[${argType}])"
-        } getOrElse {
-          q"discardEntity"
         }
       )
-    }
 
-    def headersToAkka: List[ScalaParameter] => Target[Option[Term]] = {
+    def headersToAkka: List[ScalaParameter] => Target[Option[Term]] =
       directivesFromParams(
         arg => tpe => Target.pure(q"headerValueByName(${arg})"),
         arg => tpe => Target.error(s"Unsupported Iterable[${arg}]"),
         arg => tpe => Target.error(s"Unsupported Option[Iterable[${arg}]]"),
         arg => tpe => Target.pure(q"optionalHeaderValueByName(${arg})")
       ) _
-    }
 
-    def qsToAkka: List[ScalaParameter] => Target[Option[Term]] = {
+    def qsToAkka: List[ScalaParameter] => Target[Option[Term]] =
       directivesFromParams(
         arg => tpe => Target.pure(q"parameter(Symbol(${arg}).as[${tpe}])"),
         arg => tpe => Target.pure(q"parameter(Symbol(${arg}).as[${tpe}].*)"),
         arg => tpe => Target.pure(q"parameter(Symbol(${arg}).as[${tpe}].*).map(Option.apply _)"),
         arg => tpe => Target.pure(q"parameter(Symbol(${arg}).as[${tpe}].?)")
       ) _
-    }
 
-    def formToAkka: List[ScalaParameter] => Target[Option[Term]] = {
+    def formToAkka: List[ScalaParameter] => Target[Option[Term]] =
       directivesFromParams(
         arg => tpe => Target.pure(q"formField(Symbol(${arg}).as[${tpe}])"),
         arg => tpe => Target.pure(q"formField(Symbol(${arg}).as[${tpe}].*)"),
         arg => tpe => Target.pure(q"formField(Symbol(${arg}).as[${tpe}].*).map(Option.apply _)"),
         arg => tpe => Target.pure(q"formField(Symbol(${arg}).as[${tpe}].?)")
       ) _
+
+    def asyncFormToAkka(operationId: String): List[ScalaParameter] => Target[(Option[Term], List[Stat])] = { params =>
```

Comment:
OMG 🤕

Reply:
this follows reasonably closely the strategy in akka, though widening to support more than one uploaded field: https://github.com/akka/akka-http/blob/master/akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala#L82

Reply:
I expect that consumers will want some sort of override ability to stream to S3 or something like that, which is why there are so many override points